### DISCUSSION TITLE
Got 500 system error

### DISCUSSION THREAD

**Discussion by christbiz86** [eusonlito/GPS-Tracker#76] [Q&A]
Hi, please your help. I already install and got 500 system error

![image](https://github.com/user-attachments/assets/3749bf8f-a169-4d7e-aa58-96220f672582)

**Comment by christbiz86** [comment]
the log file :
production.ERROR: App\Domains\Server\Exception\PortLocked: The server has been restarted because the 8091 is blocked in /var/www/tracker/app/Domains/Server/Action/StartPort.php:118

 production.ERROR: The "fuser" command is not available {"url":"http://localhost","method":"GET","exception":"[object] (App\\Exceptions\\UnexpectedValueException(code: 412): The \"fuser\" command is not available at /var/www/tracker/app/Services/Server/Process.php:103)

also I already install psmisc

**Comment by eusonlito** [comment]
As error says `The "fuser" command is not available` you should install `fuser`. For example with `yum install psmisc -y`

**Reply by christbiz86** [reply]
I already install fuser

![image](https://github.com/user-attachments/assets/3e632deb-e169-4902-a34f-a08775899cbd)

**Reply by eusonlito** [reply]
If you execute `type fuser` in your system, what is the response?

**Reply by christbiz86** [reply]
the response is :

fuser is /sbin/fuser

**Comment by eusonlito** [comment]
can you launch this script in your server with same PHP version?

```php
<?php var_dump(trim(strval(shell_exec('type fuser 2>/dev/null'))));
```

I get:

```
$> php fuser.php 
string(23) "fuser is /usr/bin/fuser"
```

**Reply by christbiz86** [reply]
the response is :

![image](https://github.com/user-attachments/assets/1750656c-a135-4238-a752-58583ed617df)

**Reply by eusonlito** [reply]
Can you add this code before line 102 in this file?

https://github.com/eusonlito/GPS-Tracker/blob/5c927c2258fe2d79e652db2c14d65aec6215864f/app/Services/Server/Process.php#L102-L105

```php
die(trim(strval(shell_exec('type fuser 2>/dev/null'))));
```

**Reply by christbiz86** [reply]
ok I already add that code.. and then what should I do next?

the log file still have :
production.ERROR: App\Domains\Server\Exception\PortLocked: The server has been restarted because the 8091 is blocked in /var/www/tracker/app/Domains/Server/Action/StartPort.php:118

**Reply by eusonlito** [reply]
mmm, can you check if works removing these last added line and comment the line 103: `throw new UnexpectedValueException(__('common.error.fuser-not-found'));`

**Reply by christbiz86** [reply]
it still doesn't work.

the server is centos 8 and also the local ip is 172.16.69.112 and then use port forwarding so can access from ip public.

is the ip public should be an issue?

**Reply by eusonlito** [reply]
No, is not related, can you add this lgger before line 102 in this file?

https://github.com/eusonlito/GPS-Tracker/blob/5c927c2258fe2d79e652db2c14d65aec6215864f/app/Services/Server/Process.php#L102-L105

```php
    logger()->info(['fuser' => shell_exec('type fuser')]);

     if (empty(Exec::cmd('type fuser 2>/dev/null'))) { 
         throw new UnexpectedValueException(__('common.error.fuser-not-found')); 
     } 
 }
```

I have installed the tracker in some CentOS without problem.

**Reply by christbiz86** [reply]
I already add the code and here the log : 

![image](https://github.com/user-attachments/assets/c6d9aee0-d130-4481-bb20-cc102707dfd0)

**Reply by eusonlito** [reply]
And
```
logger()->info(['fuser' => shell_exec('type fuser 2>&1')]);
```

**Reply by christbiz86** [reply]
The result is : 
![image](https://github.com/user-attachments/assets/8c3ee111-16d7-4f30-a944-bbae561a4a18)

**Reply by eusonlito** [reply]
Then, the user related with your web environment can not found the command. Is `/sbin/` inside the user `$PATH`?

```php
logger()->info(['path' => shell_exec('echo $PATH')]);
```

**Reply by christbiz86** [reply]
the user not in /sbin

![image](https://github.com/user-attachments/assets/6c89f68e-6405-4104-99b9-7ce1abfff7ff)

**Reply by eusonlito** [reply]
Then you can do (for example) a link from `/sbin/fuser` to `/usr/bin/fsuer`: `ln -s /sbin/fuser /usr/bin/fuser`.

**Reply by christbiz86** [reply]
ok and the result is : 
![image](https://github.com/user-attachments/assets/8aacf71c-9ffc-4436-bdea-304e1cc0d36f)

but I still got error 500

**Reply by eusonlito** [reply]
Ok, and the error log now is?

**Reply by christbiz86** [reply]
the error log now : 

![image](https://github.com/user-attachments/assets/f5ead422-ea76-48c0-b1ff-a63b7f029d1b)

**Reply by eusonlito** [reply]
Can you paste only the last error lines?

**Reply by christbiz86** [reply]
the last error line is : 

[2024-10-17T10:35:07+00:00] production.ERROR: App\Domains\Server\Exception\PortLocked: The server has been restarted because the 8091 is blocked in /var/www/tracker/app/Domains/Server/Action/StartPort.php:118
Stack trace:
#0 /var/www/tracker/app/Domains/Server/Action/StartPort.php(101): App\Domains\Server\Action\StartPort->killByLocked()
#1 /var/www/tracker/app/Domains/Server/Action/StartPort.php(40): App\Domains\Server\Action\StartPort->kill()
#2 /var/www/tracker/app/Domains/Core/Action/ActionFactoryAbstract.php(55): App\Domains\Server\Action\StartPort->handle()
#3 /var/www/tracker/app/Domains/Server/Action/ActionFactory.php(60): App\Domains\Core\Action\ActionFactoryAbstract->actionHandle()
#4 /var/www/tracker/app/Domains/Server/Command/StartPort.php(29): App\Domains\Server\Action\ActionFactory->startPort()
#5 /var/www/tracker/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Domains\Server\Command\StartPort->handle()
#6 /var/www/tracker/vendor/laravel/framework/src/Illuminate/Container/Util.php(43): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#7 /var/www/tracker/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(95): Illuminate\Container\Util::unwrapIfClosure()
#8 /var/www/tracker/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#9 /var/www/tracker/vendor/laravel/framework/src/Illuminate/Container/Container.php(690): Illuminate\Container\BoundMethod::call()
#10 /var/www/tracker/vendor/laravel/framework/src/Illuminate/Console/Command.php(213): Illuminate\Container\Container->call()
#11 /var/www/tracker/vendor/symfony/console/Command/Command.php(279): Illuminate\Console\Command->execute()
#12 /var/www/tracker/vendor/laravel/framework/src/Illuminate/Console/Command.php(182): Symfony\Component\Console\Command\Command->run()
#13 /var/www/tracker/vendor/symfony/console/Application.php(1047): Illuminate\Console\Command->run()
#14 /var/www/tracker/vendor/symfony/console/Application.php(316): Symfony\Component\Console\Application->doRunCommand()
#15 /var/www/tracker/vendor/symfony/console/Application.php(167): Symfony\Component\Console\Application->doRun()
#16 /var/www/tracker/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(197): Symfony\Component\Console\Application->run()
#17 /var/www/tracker/artisan(35): Illuminate\Foundation\Console\Kernel->handle()
#18 {main}  
[2024-10-17T10:35:07+00:00] production.INFO: array (
  'fuser' => 'fuser is /usr/bin/fuser
',
)  
[2024-10-17T10:35:07+00:00] production.INFO: array (
  'fuser' => 'fuser is /usr/bin/fuser
',
)  
[2024-10-17T10:35:07+00:00] production.INFO: array (
  'path' => '/usr/bin:/bin
',
)

**Reply by christbiz86** [reply]
and this one result of port 8091 

![image](https://github.com/user-attachments/assets/8da3c8d2-48e1-4fd9-af61-bfdc4b9ab61e)

**Reply by eusonlito** [reply]
Command user is `root`. Do you have root also as web user? There are maybe some permissions problem with your installation?

**Reply by christbiz86** [reply]
yes, root also as web user. is there any problem with port 8091? or is there any permission folder that I should check?

**Reply by christbiz86** [reply]
I think the issue with port 8091.. I have killed and restart but the issue still same like before @eusonlito

**Reply by alphalneo** [reply]
Probably you have php-fpm and you need to configure env[PATH] to include /usr/sbin
